src/opera/util/mock_utils.py

```diff
@@ -232,6 +232,91 @@ def GetMetadata(self):
             """
             return deepcopy(self.dummy_metadata)
 
+    class MockDistS1GdalDataset:
+        """
+        Mock class for gdal.Dataset objects, as returned from an Open call.
+        For use when mocking metadata from DIST-S1 GeoTIFF products
+        """
+
+        def __init__(self):
+            self.dummy_metadata = {
+                'apply_water_mask': 'False',
+                'bucket': 'None',
+                'dst_dir': '/home/ops/scratch_dir',
+                'high_confidence_threshold': '5.5',
+                'memory_strategy': 'high',
+                'mgrs_tile_id': '10SGD',
+                'moderate_confidence_threshold': '3.5',
+                'n_lookbacks': '3',
+                'n_workers_for_despeckling': '5',
+                'post_rtc_opera_ids': 'OPERA_L2_RTC-S1_T137-292318-IW1_20250102T015857Z_20250102T190143Z_S1A_30_v1.0,'
```

Comment:
My comment regarding the example runconfigs also sort of applies here, i.e, no need to be completely faithful to what we might see in production or a "real" test case. Just need something "representative enough" so these canned examples don't grow to be unwieldy and hard to maintain.

Reply:
Funnily enough I did. But I guess not enough (so many RTCs!)

Reply:
Pruned it down to 5 each like with the example RC